### PR TITLE
Enable s3 for f6

### PR DESCRIPTION
**Pull request by emudojo**

### CODE REVIEW

File: fcrepo6/rootfs/opt/tomcat/bin/setenv.sh [hunk 1]
```diff
@@ -32,3 +32,14 @@ fi
...
+
+if [[ "${FCREPO_BINARYSTORAGE_TYPE}" == "file" ]]; then
+  export CATALINA_OPTS="${CATALINA_OPTS} -Dfcrepo.storage=ocfl-fs"
+fi
+if [[ "${FCREPO_BINARYSTORAGE_TYPE}" == "s3" ]]; then
+  # Export AWS credentials
+  export AWS_ACCESS_KEY_ID=${FCREPO_S3_USER}
+  export AWS_SECRET_ACCESS_KEY=${FCREPO_S3_PASSWORD}
+  # Enable S3 mode and set default options
+  export CATALINA_OPTS="${CATALINA_OPTS} -Dfcrepo.storage=ocfl-s3 -Dfcrepo.aws.region=${FCREPO_AWS_REGION} -Dfcrepo.aws.region=${FCREPO_S3_BUCKET} -Dfcrepo.aws.region=${FCREPO_S3_PREFIX}"
```
Comment (nigelgbanks) [inline fcrepo6/rootfs/opt/tomcat/bin/setenv.sh:44]: Should `-Dfcrepo.aws.region` be specified with three different values (`FCREPO_AWS_REGION`, `FCREPO_S3_BUCKET` and `FCREPO_S3_PREFIX`) or should each of the environment variables be passed to a different Java property?

**Review by nigelgbanks — APPROVED** [review]

**Comment by nigelgbanks** [comment]
I tried to get this working with minio (see the attached tests), with the hopes of doing some automated testing, but couldn't get it working. That being said it seems to be more so an issue with Fedora and how it's configured to assume s3 and not some s3 compliant API which might explain that. 

Additionally the output doesn't seem to match up in that it thinks no access key / secret key is set? Though this again may be some weirdness around Fedora. They do not really document it directly here: https://wiki.lyrasis.org/display/FEDORA6x/Amazon+S3, but rather point to the AWS documentation: https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html. Which according to that the implementation here is correct so I don't know what to make of that.

```
fcrepo_1    | INFO 11:49:46.259 (OcflPropsConfig) Fedora AWS access key: 
fcrepo_1    | INFO 11:49:46.259 (OcflPropsConfig) Fedora AWS secret key set: false
fcrepo_1    | INFO 11:49:46.259 (OcflPropsConfig) Fedora AWS region: us-west-2
fcrepo_1    | INFO 11:49:46.259 (OcflPropsConfig) Fedora OCFL S3 bucket: minio.traefik.me/fcrepo
fcrepo_1    | INFO 11:49:46.259 (OcflPropsConfig) Fedora OCFL S3 prefix: 
fcrepo_1    | INFO 11:49:46.259 (OcflPropsConfig) Fedora OCFL digest algorithm: SHA-512
```

That being said I haven't tried this against an actual s3 bucket with credentials, but I'm happy to merge this as is, but will leave it for a day or so for anyone else who can / wants to test on an actual s3 bucket. 


[test.zip](https://github.com/Islandora-Devops/isle-buildkit/files/6271341/test.zip)

**Comment by nigelgbanks** [comment]
@emudojo is this ready to be merged or are you still testing against s3?

**Comment by elizoller** [comment]
is there anything holding up this PR?

**Comment by noahwsmith** [comment]
I'll get this back on my radar to test/tweak/merge

**Comment by noahwsmith** [comment]
Ok, to Nigel's question we're still waiting on Minio support on the Fedora side. The next time they cut a release we can work in a custom endpoint variable to support this. This work has been done here: https://github.com/fcrepo/fcrepo/pull/1894

The other GOTCHA to know about is that you have to use Postgres as your FCCREPO_DB_HOST if you want this to work. That's also just a function of how F6 works right now.

We've been using this Fedora6 image on our Whitman/Lyrasis/IMLS project with no issues for a few weeks now and it works great. 

Here are the `.env` variables we use:

```
# FCREPO_TAG=d63e8563355b21457a56a8d427af97b82215749e
# FCREPO_BINARYSTORAGE_TYPE=s3
# FCREPO_PERSISTENCE_TYPE=postgresql
# FCREPO_DB_HOST=postgresql
# FCREPO_DB_PORT=5432
# FCREPO_S3_BUCKET=abc123-test
# FCREPO_S3_USER=AK123...
# FCREPO_S3_PASSWORD=1J...
# FCREPO_S3_PREFIX=
# FCREPO_AWS_REGION=us-east-1
```

I'm not sure what we need to do to get some of this into isle-dc (pinging @nigelgbanks on that front), but I believe this PR can be accepted as-is.

**Comment by nigelgbanks** [comment]
I've merged it, my only concern was that it had been tested, the changes here should not cause any existing functionality to break.

**Comment by nigelgbanks** [comment]
@noahwsmith there doesn't appear to be any pull request against isle-dc for changes, I suppose documentation is the only thing required since this use case won't be the default. Though that being said, since this isn't fully fleshed out and is waiting on changes to F6, might as well just wait till it all works before documenting it to reduce the effort.

**Pull request merged**